docs/advanced/developer-guide.md

```diff
@@ -0,0 +1,303 @@
+
+```bash
+$ docker run --rm --name=nfd-test <NFD_CONTAINER_IMAGE> nfd-master --no-publish
+2019/02/01 14:48:21 Node Feature Discovery Master <NFD_VERSION>
+2019/02/01 14:48:21 gRPC server serving on port: 8080
+```
+
+Command line flags of nfd-master:
+
+```bash
+$ docker run --rm <NFD_CONTAINER_IMAGE> nfd-master --help
+...
+Usage:
+  nfd-master [--prune] [--no-publish] [--label-whitelist=<pattern>] [--port=<port>]
+     [--ca-file=<path>] [--cert-file=<path>] [--key-file=<path>]
+     [--verify-node-name] [--extra-label-ns=<list>] [--resource-labels=<list>]
+     [--kubeconfig=<path>]
+  nfd-master -h | --help
+  nfd-master --version
+
+  Options:
+  -h --help                       Show this screen.
+  --version                       Output version and exit.
+  --prune                         Prune all NFD related attributes from all nodes
+                                  of the cluster and exit.
+  --kubeconfig=<path>             Kubeconfig to use [Default: ]
+  --port=<port>                   Port on which to listen for connections.
+                                  [Default: 8080]
+  --ca-file=<path>                Root certificate for verifying connections
+                                  [Default: ]
+  --cert-file=<path>              Certificate used for authenticating connections
+                                  [Default: ]
+  --key-file=<path>               Private key matching --cert-file
+                                  [Default: ]
+  --verify-node-name              Verify worker node name against CN from the TLS
+                                  certificate. Only has effect when TLS authentication
+                                  has been enabled.
+  --no-publish                    Do not publish feature labels
+  --label-whitelist=<pattern>     Regular expression to filter label names to
+                                  publish to the Kubernetes API server.
+                                  NB: the label namespace is omitted i.e. the filter
+                                  is only applied to the name part after '/'.
+                                  [Default: ]
+  --extra-label-ns=<list>         Comma separated list of allowed extra label namespaces
+                                  [Default: ]
+  --resource-labels=<list>        Comma separated list of labels to be exposed as extended resources.
+                                  [Default: ]
+```
+
+### NFD-Worker
+
+In order to run nfd-worker as a "stand-alone" container against your
+standalone nfd-master you need to run them in the same network namespace:
+
+```bash
+$ docker run --rm --network=container:nfd-test <NFD_CONTAINER_IMAGE> nfd-worker
+2019/02/01 14:48:56 Node Feature Discovery Worker <NFD_VERSION>
+...
+```
+
+If you just want to try out feature discovery without connecting to nfd-master,
+pass the `--no-publish` flag to nfd-worker.
+
+Command line flags of nfd-worker:
+
+```bash
+$ docker run --rm <CONTAINER_IMAGE_ID> nfd-worker --help
```

Comment:
Shouldn't the CONTAINER_IMAGE_ID be the same as above NFD_CONTAINER_IMAGE for consistency?

Reply:
Yes, changed